### PR TITLE
Add ActivitySource/OpenTelemetry support

### LINKED ISSUE #996: Add support for OpenTelemetry
It will be nice to have more insights directly from the library and enable users to integrate easily with [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-dotnet). 
The `Step` and `Workflow` middleware are not enough to get full traceability what happens into the library.

If it's ok, I will do a PR to integrate [ActivitySource](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/distributed-tracing-instrumentation-walkthroughs) into the library.

### PR DESCRIPTION
**Pull request by glucaci**
Fixes #996

- add support for configuring `IdleTime`
- add base support for tracing with `ActivitySource`

When using [opentelementry](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started/README.md) to instrument a service is nice to have any incoming/outgoing request grouped under a root activity with some meaningful naming.

With this PR I've tried to add some minimal support to have everything covered. I'm using opentelemetry-dotnet with opentelemetryprotocol exporter into elastic.

Here are some sample traces before and after this PR:
### Traces generated by runnable pollers, queue consumers and start process
image

For example under `workflow start host` you can see now all HTTP requests to AzureStorage oder other provider
image

### Traces generated by running a workflow
image

Here the traces are splited by step because each step run is independent.

### CODE REVIEW

**Comment by glucaci** [comment]
@danielgerlag any thoughts on this?

**Comment by danielgerlag** [comment]
@glucaci sorry for the delay, I will start look into this in the next few days.

File: src/WorkflowCore/Models/WorkflowOptions.cs [hunk 1]
```diff
@@ -75,6 +75,11 @@ public void UseErrorRetryInterval(TimeSpan interval)
             ErrorRetryInterval = interval;
         }
 
+        public void UseIdleTime(TimeSpan interval)
```
Comment (danielgerlag) [inline src/WorkflowCore/Models/WorkflowOptions.cs:78]: Could we name this to be more specific about where the idle is happening?

File: src/WorkflowCore/WorkflowCore.csproj [hunk 2]
```diff
@@ -26,6 +26,7 @@
     <PackageReference Include="System.Reflection.TypeExtensions" Version="4.7.0" />
     <PackageReference Include="System.Threading.Thread" Version="4.3.0" />
     <PackageReference Include="System.Linq.Queryable" Version="4.3.0" />
+    <PackageReference Include="OpenTelemetry.Api" Version="1.2.0-rc2" />
```
Comment (danielgerlag) [inline src/WorkflowCore/WorkflowCore.csproj:29]: Taking a dependency on pre-release packages could be a blocker for some organizations.

**Review by danielgerlag — COMMENTED** [review]
Nice work

File: src/WorkflowCore/WorkflowCore.csproj (hunk 2 above)
Comment (glucaci) [reply src/WorkflowCore/WorkflowCore.csproj]: downgraded to last stable version

**Comment by glucaci** [comment]
Hei @danielgerlag do you think we could merge this soon?

**Comment by danielgerlag** [comment]
The build seems to be failing

**Comment by glucaci** [comment]
@danielgerlag  the build is failing because of the samples.
I did a PR for updating the samples and test TF #1021

**Comment by glucaci** [comment]
@danielgerlag seams that no EntityFramework provider is working with .NET6
I've created a PR to fix them #1023

After it's merged this PR should also have a green build.

**Pull request merged**